tuplex/core/src/ee/local/LocalBackend.cc

```diff
@@ -1528,8 +1558,14 @@ namespace tuplex {
         }
 #endif
 
+        for (int i = 0; i < tasks.size(); i++) {
+            // take limit only work with uniform order
```

Comment:
this should then be within an if block checking for hasOutputLimit,  no?

Reply:
Good point! I moved this to `executeTransformStage` and add the if check